src/SignalR/clients/csharp/Http.Connections.Client/src/Internal/WebSocketsTransport.cs

```diff
@@ -142,9 +151,8 @@ public async Task StartAsync(Uri url, TransferFormat transferFormat, Cancellatio
                     }
                     else
                     {
-#pragma warning disable CA1416 // Analyzer bug
-                        _webSocket.Options.SetRequestHeader("Authorization", $"Bearer {accessToken}");
-#pragma warning restore CA1416 // Analyzer bug
+                        _httpConnectionOptions.Headers.Remove("Authorization");
```

Comment:
yeah, it totally makes sense. i also don't like deep copy that much. and also we can't just use `HttpConnectionOptions.AccessTokenProvider` from the options itself since `HttpConnection` wraps it in this method https://github.com/dotnet/aspnetcore/blob/8cf98856bb96c7aad8373ae5b1fa6e7f021f649e/src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnection.cs#L647-L654 and returns null string if no AccessTokenProvider is specified. 
If we would use `AccessTokenProvider` directly from `HttpConnectionOptions`, then it won't have that wrapped version. And from my understanding we can't change it in options object as well as it's user provided options.

i did another way by adding `public string? AccessToken {get;}` to `WebSocketConnectionContext`. So `WebSocketsTransport` still uses wrapped AccessTokenProvider and gets the actual token, and then it just passes token to context and `WebSocketFactory` would just use it to pass as header\queryParam. Would it be fine or there is a better way of doing that?

Reply:
You can set the new AccessTokenProvider on the HttpConnectionOptions in WebSocketTransport, it wont affect the users options because it's a property on the WebSocketTransport specific options class not the original options class passed in by the user.

Reply:
oh, i totally missed that. since options are copying on the transport factory stage we can replace that property since it's just a delegate. and it doesn't work for Dictionary since we maintain original instance there.
pushed update